tensorflow/core/kernels/data/concatenate_dataset_op.cc

```diff
@@ -51,9 +51,11 @@ class ConcatenateDatasetOp::Dataset : public DatasetBase {
 
     auto os_input = input->output_shapes();
     auto os_concatenate = to_concatenate->output_shapes();
+    PartialTensorShape output_tensorshape({});
```

Comment:
Just needs to move inside the for loop, otherwise every new iteration would start from the result of the previous one

Reply:
Yes I had it locally.. now pushed